src/bot.coffee

```diff
@@ -173,7 +173,9 @@ class SlackBot extends Adapter
 
       when 'message', 'bot_message'
         @robot.logger.debug "Received message: '#{text}' in channel: #{channel.name}, from: #{user.name}"
-        @receive new TextMessage(user, text, message.ts)
+        textMessage = new TextMessage(user, text, message.ts)
+        textMessage.thread_ts = message.thread_ts
```

Comment:
I will admit, I don't love this. However since the adapter now uses the build in types from `hubot` proper I'm not sure how else to ensure we have this critical piece of data